app/provider/app.go

```diff
@@ -470,12 +471,15 @@ func New(
 		app.BankKeeper,
 		scopedTransferKeeper,
 	)
-	transferModule := transfer.NewAppModule(app.TransferKeeper)
-	ibcmodule := transfer.NewIBCModule(app.TransferKeeper)
+
+	// Add an IBC middleware callback to track the consumer rewards
+	var transferStack porttypes.IBCModule
+	transferStack = transfer.NewIBCModule(app.TransferKeeper)
+	transferStack = provider.NewIBCMiddleware(transferStack, app.ProviderKeeper)
```

Comment:
Need to be properly tested on Gaia as the transfer stack is much larger. @sainoe we should have upgrading instructions on how to integrate this and if there are any constraints on the order in the stack.

Reply:
Agree. I will create an issue to make sure we test the transfer stack in Gaia.